doc/code-style.rst

```diff
@@ -0,0 +1,734 @@
+  by the indent level.
+
+- Use region markers and separators when appropriate. Region markers consist of
+  C++ comments followed *immediately* (i.e. with no intervening whitespace) by
+  ``BEGIN`` or ``END``, followed by a brief description of the contents of the
+  region. Regions should be separated from one another by a line consisting of
+  79 forward slashes surrounded by blank lines. If appropriate, a region
+  separator may also separate an unmarked region at the beginning or end of a
+  file from an adjacent annotated region. Region begin/end markers should
+  always appear in pairs.
+
+  .. admonition:: Rationale:
+
+    Similar to header lines, region markers can help a reader quickly locate
+    their place within a file, and also serve to split larger files into
+    smaller chunks of related functionality. The specific syntax here described
+    is recognized by some editors and will be specially highlighted and may
+    create folding regions.
+
+- Comments (meaning here, *textual* comments, not region markers and separator
+  lines) should be used judiciously. Don't use comments to say something that
+  is obvious from the code itself, but do use comments to point out design
+  choices, "gotchas", areas that need improvement, and to help separate blocks
+  of related code.
+
+- Comments should start with a capital letter, and should use correct spelling
+  and grammar. (If possible, use an editor with built-in spell checking.)
+  However, comments normally do *not* end with a period, unless a comment
+  consists of more than one sentence. Sentence fragments (as long as the
+  grammar is not atrocious) are acceptable.
+
+- When using C-style multi-line comments, the initial :cpp:`/*` should be
+  indented to the same level as surrounding code. Additional lines should start
+  with :code:`*` and be indented one additional space, so that each :code:`*`
+  lines up with the *first* :code:`*` of the initial line. (However, prefer to
+  avoid such comments aside from copyright notices and Doxygen documentation
+  blocks. For that matter, consider using :cpp:`///` for Doxygen documentation
+  of new code.)
+
+- Function prototypes should place the return type, class name, and method name
+  on separate lines. Parameters may start on the same line as the method name.
+
+  .. admonition:: Example:
+
+    .. code:: c++
+
+      result_t
+      some_class
+      ::some_method( int the_param ) const
+      {
+        ...
+      }
+
+- A :cpp:`:` following a class declaration or constructor should be indented
+  and preceded by a newline, if the preceding and following text is not all on
+  the same line. (Base class lists, however, may span lines without breaking
+  before the :cpp:`:`.)
+
+  .. admonition:: Example:
+
+    .. code:: c++
+
+    // Okay
+    class my_class : public really_long_name_of_base_class,
+                     protected another_really_long_class_name
```

Comment:
I am trying to read this in my rst view in pycharm and it will not render this due to the indentation